src/rustfmt_diff.rs

```diff
@@ -164,6 +179,45 @@ where
     }
 }
 
+/// Convert a Mismatch into a serialised form which just includes
+/// enough information to modify the original file.
+/// Each section starts with a line with three integers, space separated:
+///     lineno num_removed num_added
+/// followd by (num_added) lines of added text.  The line numbers are
+/// relative to the original file.
+pub fn output_modified<W>(mut out: W, diff: Vec<Mismatch>)
```

Comment:
Do we need this function? I thought you removed the facility to write out the 'modified' output

Reply:
Well, it's used to serialise the result (really just line count header lines before the changed lines) into the `out` passed into `write_file`.  The alternatives (either of which I'm happy to do) are adding another parameter to `write_file` or coming up with another similar function which had something else instead of the `out: &mut T` to return the modified lines more directly.